### PR TITLE
Initialize Storybook React

### PR DESCRIPTION
**Pull request by garrettbear**
## **Description**

This PR introduces a starter setup for `docs/storybook-react` and configures Tailwind CSS to work with Storybook. We set up the `packages/react-components` as a placeholder for testing components within Storybook and successfully integrated MetaMask's design tokens into the Storybook environment. 

Key changes:
- Created a starter for `docs/storybook-react`.
- Configured Tailwind CSS in Storybook (excluding `tailwind-preset`, which is handled in a separate ticket).
- Added `packages/react-components` as a placeholder to facilitate testing in Storybook.
- Integrated MetaMask design tokens to ensure consistent styling within the Storybook environment.
- Cleaned up most of the Storybook starter files, leaving a few necessary ones for future configurations.

## **Related issues**

Fixes: #729

## **Manual testing steps**

1. Navigate to `docs/storybook-react`.
2. Run `yarn storybook` to start the Storybook environment.
3. Verify that the `packages/react-components` placeholder components are loading correctly in Storybook.
4. Check that the Tailwind utilities are working as expected in the components.
5. Confirm that design tokens are being applied properly to the components.
6. Ensure Storybook is functional with the current configuration and no missing files.

### **Before**

N/A

### **After**

https://github.com/user-attachments/assets/43b2cc5e-c02b-420c-93dd-65f12d88a1b5

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs).
- [x] I've completed the PR template to the best of my ability.
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g., pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by georgewrmarshall** [comment]
@SocketSecurity ignore-all

File: docs/storybook-react/.storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|ts|tsx)',
+    '../../../packages/react-components/src/**/*.stories.@(js|jsx|ts|tsx)',],
+
+  addons: [
+    getAbsolutePath('@storybook/addon-onboarding'),
+    getAbsolutePath('@storybook/addon-links'),
+    getAbsolutePath('@storybook/addon-essentials'),
+    getAbsolutePath('@chromatic-com/storybook'),
+    getAbsolutePath('@storybook/addon-interactions'),
+    getAbsolutePath('@storybook/addon-a11y'),
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply docs/storybook-react/.storybook/main.ts:26]: thanks for the context

File: docs/storybook-react/.storybook/theme.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { create } from '@storybook/theming';
+
+export default create({
+  base: 'light',
+  brandTitle: 'MetaMask Design System React Components',
+  brandUrl: 'https://metamask.io/',
+  brandImage:
+    'https://images.ctfassets.net/9sy2a0egs6zh/4zJfzJbG3kTDSk5Wo4RJI1/1b363263141cf629b28155e2625b56c9/mm-logo.svg',
+  brandTarget: '_blank',
+});
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply docs/storybook-react/.storybook/theme.ts:10]: good to keep for now until we decide whatever other logo to put in there instead

File: docs/storybook-react/tailwind.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (brianacnguyen) [inline docs/storybook-react/tailwind.css:3]: maybe put a comment here to remind us to also import design-tokens settings as well?

File: docs/storybook-react/tsconfig.app.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    /* Bundler mode */
+    "moduleResolution": "bundler",
+    "allowImportingTsExtensions": true,
+    "isolatedModules": true,
+    "moduleDetection": "force",
+    "noEmit": true,
+    "jsx": "react-jsx",
+
+    /* Linting */
```
Comment (brianacnguyen) [inline docs/storybook-react/tsconfig.app.json:17]: i dont think comments are allowed in json files

**Review by brianacnguyen — COMMENTED** [review]
LGTM. a few changes needed tho

**Pull request closed without merging**